components/director/internal/domain/formationassignment/notifications.go

```diff
@@ -401,7 +401,7 @@ func convertFormationAssignmentFromModel(formationAssignment *model.FormationAss
 		return &webhook.FormationAssignment{Value: "\"\""}
 	}
 	config := string(formationAssignment.Value)
-	if config == "" {
+	if config == "" || formationAssignment.State == string(model.CreateErrorAssignmentState) || formationAssignment.State == string(model.DeleteErrorAssignmentState) {
```

Comment:
Shouldn't we also adapt the state as well? In case of "create_error" to set the state in "initial" and when we have "delete_error" --> "deleting"?

P.S. Valid for the other changes as well

Reply:
This is during the sending of the notifications. The transition is already done after it is sent in `CleanupFormationAssignment` and `processFormationAssignmentsWithReverseNotification`. 
The flow is as follows:
1. Get all assignments
2. Build notifications payload
3. Send notification 
4. Update the Formation Assignment based on the response (and in the case from CREATE_ERROR/DELETE_ERROR to INITIAL/DELETING it is cleared)

Reply:
There are even e2e tests that currently check the cleanup of errors when changing to `DELETING` and `INITIAL`, but there are missing asserts for the notifications that are received by external service mock.